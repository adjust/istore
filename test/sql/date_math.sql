SELECT '2012-01-02'::aj_date + 1;
SELECT 1 + '2012-01-02'::aj_date;
SELECT '2012-01-02'::aj_date - 1;
SELECT '2012-01-02'::aj_date - 2;
SELECT '2012-01-02'::aj_date + 100000;
SELECT '2012-01-02'::aj_date - '2012-01-01'::aj_date;
SELECT '2012-01-01'::aj_date - '2012-01-02'::aj_date;
