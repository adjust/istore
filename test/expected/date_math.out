SELECT '2012-01-02'::aj_date + 1;
  ?column?  
------------
 2012-01-03
(1 row)

SELECT 1 + '2012-01-02'::aj_date;
  ?column?  
------------
 2012-01-03
(1 row)

SELECT '2012-01-02'::aj_date - 1;
  ?column?  
------------
 2012-01-01
(1 row)

SELECT '2012-01-02'::aj_date - 2;
ERROR:  resulting date is out of aj_date range
SELECT '2012-01-02'::aj_date + 100000;
ERROR:  resulting date is out of aj_date range
SELECT '2012-01-02'::aj_date - '2012-01-01'::aj_date;
 ?column? 
----------
        1
(1 row)

SELECT '2012-01-01'::aj_date - '2012-01-02'::aj_date;
 ?column? 
----------
       -1
(1 row)

